platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -107,40 +104,23 @@ public ConfigurationUpdater(
   }
 
   /**
-   * @inheritDoc
-   * @throws IOException if an error occurs reading configuration data
-   * @throws InvalidSyntaxException technically impossible since {@code null} is being passed as the
-   *     configuration filter. See {@link ConfigurationAdmin#listConfigurations(String)}.
+   * @inheritDoc Initialize the updater with all the currently known configuration.
    * @see ConfigurationPersistencePlugin
    */
   @Override
-  public void initialize(ConfigurationContextFactory factory) {
-    Configuration[] configs;
-    try {
-      configs = ddfConfigAdmin.listConfigurations(null);
-    } catch (IOException | InvalidSyntaxException e) {
-      LOGGER.error("Problem fetching configurations. ", e);
-      return;
-    }
-    if (configs == null) {
-      return;
-    }
-    Arrays.stream(configs)
-        .map(factory::createContext)
-        .forEach(
-            context -> {
-              try {
-                handleStore(context);
-              } catch (IOException e) {
-                LOGGER.error(
-                    "Problem updating config file [{}]. {}",
-                    context.getConfigFile(),
-                    e.getMessage());
-                if (LOGGER.isDebugEnabled()) {
-                  LOGGER.debug("Exception occurred while trying to updat config file. ", e);
-                }
-              }
-            });
+  public void initialize(List<ConfigurationContext> state) {
+    state.forEach(
+        context -> {
+          try {
+            handleStore(context);
+          } catch (IOException e) {
+            LOGGER.error(
+                "Problem updating config file [{}]. {}", context.getConfigFile(), e.getMessage());
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
❓ Why the if(), you should be fine to simply call debug()
But this can be done in one call to error() by using the error(String, Throwable) version

Reply:
I'm trying to discourage the trend of throwing stack traces in error messages. Those should be reserved for debug. We do this everywhere and it makes the logs ugly.

Reply:
👍 for the debug but you still do not need the if before it